zaza/openstack/charm_tests/vault/utils.py

```diff
@@ -264,6 +264,16 @@ def is_initialized(client):
     return client.hvac_client.is_initialized()
 
 
+# auto-retry this because this is called immediately after
+# unsealing vault, and vault can take some time to be ready.
+@tenacity.retry(
+    retry=tenacity.retry_if_exception_type((
+        hvac.exceptions.InternalServerError,  # raft cluster may be settling
+        hvac.exceptions.VaultDown, # vault may still be unsealing
+    )),
+    reraise=True,
+    wait=tenacity.wait_fixed(5),
+    stop=tenacity.stop_after_attempt(10))
```

Comment:
I'd much rather actually check that vault is ready rather than failing and relying on timeouts.  This makes it obvious that the issue isn't, say, slow hardware or a regression in vault/charm, rather than it simply not working.